support api token for pingdom metric provider

davidmontoyago opened the pull request:
Pingdom's new API version favors API tokens over credentials. This PR adds support for using pingdom API tokens.

StatusPage UI yields this warning if still using creds:

```
A new API version for Pingdom requires you to reauthenticate before December 1, 2020.
```

yannh commented:
It does break for whoever still uses credentials though. Maybe we shouldnt fail (yet) if APIToken is not defined?

davidmontoyago commented:
Makes sense! I'll work on a revision to allow for both cases - coming soon!

davidmontoyago commented:
@yannh I've added https://github.com/yannh/statuspage-go-sdk/pull/5 to check for the pingdom credentials only when the API token is not provided. On this PR I've updated the docs/examples to document the support of API token. I've confirmed that when using a Pingdom API token, no email is required (I've verified this as well on statuspage UI).

I tried to print WARNINGS when using pingdom creds instead of the API token (given than [statuspage support for creds ends on Dec 2020](https://support.atlassian.com/statuspage/docs/update-your-pingdom-account/)) but it wasn't as simple given the limitation described on https://github.com/hashicorp/terraform-plugin-sdk/issues/88. In any case, once statuspage stops supporting Pingdom creds, an error will be displayed to users.
